uportal-war/src/main/java/org/jasig/portal/groups/db/ExternalDBGroupStore.java

```diff
@@ -0,0 +1,345 @@
+                group.setName((String) groupDef.get("group_name"));
+                group.setDescription((String) groupDef.get("group_name"));
+                newGroups.put(group.getName(), group);
+            }
+            
+            synchronized (groups) {
+                groups.clear();
+                groups.putAll(newGroups);
+            }
+            log.warn(groups);
+    }
+    
+    @Override
+    public boolean contains(IEntityGroup group, IGroupMember member) throws GroupsException {
+        log.warn("Contains ?  "+group.getKey() + " : "+member.getKey());
+        if (member.getEntityType() != IPerson.class) return false;
+
+        return !jdbcTemplate.queryForList(groupMemberQuery, group.getName(), member.getKey()).isEmpty();
+//        int rows = .size();
+//        return rows > 0;
+        
+    }
+
+    @Override
+    public void delete(IEntityGroup group) throws GroupsException {
+        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
+    }
+
+    @Override
+    public IEntityGroup find(String key) throws GroupsException {
+        if (key.equals(ROOT_GROUP.getLocalKey())) return ROOT_GROUP;
+        return groups.get(key);
+        
+    }
+
+    @Override
+    public Iterator findContainingGroups(IGroupMember gm) throws GroupsException {
+        log.warn("Finding Containing Groups of "+gm.getKey());
+        List<IEntityGroup> rslt = new LinkedList<IEntityGroup>();
+        if (gm.isGroup()) {
+            IEntityGroup group = (IEntityGroup) gm;
+//            if (groups.containsKey(group.getLocalKey())) rslt.add(ROOT_GROUP);
+            if (groups.containsKey(group.getLocalKey())) rslt.add(ROOT_GROUP);
+        } else if (gm.isEntity() && gm.getEntityType().equals(ROOT_GROUP.getEntityType())) {	
+            
+            List<Map<String, Object>> groupMemberships = jdbcTemplate.queryForList(groupMembershipQuery, gm.getKey());
+            for (Map<String,Object> row : groupMemberships) {
+                if (groups.containsKey(row.get("group_name")))
+                    rslt.add(groups.get(row.get("group_name")));
+            }
+        } 
+        log.warn(rslt);
+        return rslt.iterator();
+    }
+
+    @Override
+    public Iterator findEntitiesForGroup(IEntityGroup group) throws GroupsException {
+        log.warn("entities ?  "+group.getLocalKey());
+        if (groups.containsKey(group.getLocalKey())) {
+
+            List<Map<String, Object>> rows = jdbcTemplate.queryForList(groupEntityMembersQuery, group.getName());
+
+            List<IEntity> entities = new ArrayList<IEntity>();
+            for (Map<String, Object> row : rows) {
+                IEntity entity = newInstance((String) row.get("username"), IPerson.class);
+                entities.add(entity);
+            }
+            
+            return entities.iterator();
+        } else {
+            return findMemberGroups(group);
```

Comment:
I'm not sure this is what's supposed to happen (though I haven't read the method yet).  I think findEntitiesForGroup is supposed to return only non-group members.